homework_oop/src/homework_oop/BusNetwork.java

```diff
@@ -0,0 +1,94 @@
+package homework_oop;
+
+public class BusNetwork extends Network implements Functionable {
+	private Server server;
+
+	public BusNetwork(String name, Server server) {
+		super(name);
+		this.server = server;
+	}
+
+
+	/**
+	 * Add computer in the computer array network.
+	 * @param pc - Computer that needs to be added.
+	 */
+	public void addComputer(Computer computer) {
+
+		Client c = (Client) computer;
+
+		Computer[] computers = new Computer[getComputers().length + 1];
+
+		// Checks if the network is empty
+		if (getComputers()[0] == null) {
+			computers[0] = c;
+			c.connect(this);
+		} else {
+			System.arraycopy(getComputers(), 0, computers, 0, getComputers().length);
+			computers[computers.length - 1] = c;
+			c.connect(this);
+			setComputers(computers);
+		}
+
+	}
+
+	/**
+	 * Remove computer from the computer network.
+	 * @param pc - Computer that needs to be removed.
+	 */
+	public void removeComputer(Computer computer) {
+
+		Client c = (Client) computer;
+
+		for (int i = 0; i < getComputers().length; i++) {
+			if (c.equals(getComputers()[i])) {
+				c.disconnect();
+				getComputers()[i] = null;
+			} else {
+				throw new IllegalArgumentException();
+			}
+		}
+
+		Computer[] computers = new Computer[getComputers().length - 1];
```

Comment:
This method will not work correctly. You are iterating up the size of the new array which is smaller than the original one. In case computer that was not last in the original list was removed, you will not copy all the remaining computers correctly into the new array.